Modified implementation to support instance tag values to maximum of 0xf...

cobratbq opened the pull request:
...fffffff, which is the maximum value allowed by OTRv3.

Additionally, added some tests to ensure that maximum value is honored
even though Java interprets these as negative values, as the last bit is
the sign bit for Java's signed int primitive.
Had to change to random value generation, since this was limited by an
int > 0 parameter.

cobratbq committed:
Modified implementation to support instance tag values to maximum of 0xffffffff, which is the maximum value allowed by OTRv3.

Additionally, added some tests to ensure that maximum value is honored
even though Java interprets these as negative values, as the last bit is
the sign bit for Java's signed int primitive.
Had to change to random value generation, since this was limited by an
int > 0 parameter.

cobratbq commented:
See issue https://github.com/otr4j/otr4j/issues/51.

cobratbq commented:
The CI build failed on a timeout of a (different) test.

```
Running net.java.otr4j.test.SMPTest
Tests run: 84, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 89.812 sec <<< FAILURE! - in net.java.otr4j.test.SMPTest
goodPasswordNoQuestion[0](net.java.otr4j.test.SMPTest)  Time elapsed: 2.062 sec  <<< ERROR!
java.lang.Exception: test timed out after 2000 milliseconds
```

eighthave commented:
Overall makes sense.  Here are my comments:
- Wouldn't `Integer.MAX_VALUE` be a more appropriate `HIGHEST_VALUE`?  That is not `0xffffffff`, it is actually `0x7fffffff` since integer types are always signed in Java.
- This doesn't seem right, shouldn't it be `0 > tagValue`?

``` diff
-   public static boolean isValidInstanceTag(int tagValue) {
-       return  tagValue == 0 ||
-               (tagValue >= SMALLEST_VALUE && tagValue <= HIGHEST_VALUE);
+   public static boolean isValidInstanceTag(final int tagValue) {
+       return !(0 < tagValue && tagValue < SMALLEST_VALUE);
    }
```
- about the timeout problem, see #48

eighthave commented:
Ok duh, I read your description too quickly.  It would be useful to add a comment in declaration of `HIGHEST_VALUE` about why `0xffffffff` and not `Integer.MAX_VALUE`.

eighthave commented:
I think `isValidInstanceTag()` should also have a comment describing that is interpreting the negative numbers as `0x80000000` through `0xffffffff`.

cobratbq committed:
Added more clarifications on how the upper range of the int primitive type works for Instance Tag values.

The 0x80000000 through 0xffffffff are interpreted as negative, because
Java int is signed. This should now be sufficiently clarified in the
comments.

cobratbq commented:
@eighthave I added some more clarifications. It should be okay now.
